typed-store/src/lib.rs

```diff
@@ -69,11 +87,28 @@ where
                             }
                         }
                     }
-
+                    StoreCommand::DeleteAll(keys, sender) => {
+                        let response = keyed_db.multi_remove(keys.iter());
+                        // notify the obligations only when the delete was successful
+                        if response.is_ok() {
+                            for key in keys {
+                                if let Some(mut senders) = obligations.remove(&key) {
+                                    while let Some(s) = senders.pop_front() {
+                                        let _ = s.send(Ok(None));
+                                    }
+                                }
+                            }
+                        }
```

Comment:
I am not sure that there is value doing this . Based on the implementation of the `notify_read` the obligation:
* either will be notified immediately if the value is available
* or will be notified once it becomes available

in order for an obligator to be notified when a value gets deleted we should have a value already in place, which means that the obligator will have already received a notification via the earlier `write` operation and then get popped up by the queue. 

I might be missing a case here, but if we don't see this possible I would lean towards removing this code

Reply:
The goal of the `notify_read` operation is indeed to implement a post-write callback signal. 

The use case is to make sure you only move on when you have guarantees you'll be able to recover from a process crash through persistence, and in practice we use it as a barrier (i.e. you have X writes to perform and only move on when they've all been done). I've never seen the value of a `notify_read` actually used or inspected, but rather always discarded: the signal is we're *done* with the next write operation on those keys, rather than the details of the written semantics.

Does that make things more clear? I agree that the name `notify_read` and the discipline of actually returning the value when we don't use it are confusing, and that there are probably simplifications to be found there. But I think discharging an obligation on a delete makes sense to me.

Reply:
Yeah, I guess looking on the code of `notify_read` it seemed to me that the caller will want to get notified _once the data are written by someone (becoming available)_ ....to give an analogy, like a one-off pub/sub with persistence. Ok cool, I'll leave as is then, but it seems that we might need to iterate this in the future , but probably not a priority for now.